module/VuFind/src/VuFind/Db/Row/User.php

```diff
@@ -576,19 +555,13 @@ public function deleteLibraryCard($id)
             throw new \VuFind\Exception\LibraryCard('Library Cards Disabled');
         }
 
-        $userCard = $this->getDbTable('UserCard');
-        $row = $userCard->select(['id' => $id, 'user_id' => $this->id])->current();
-
-        if (empty($row)) {
-            throw new \Exception('Library card not found');
-        }
-        $row->delete();
+        $row = $this->getUserCardService()->deleteLibraryCard($this->id, $id);
```

Comment:
It feels a little weird to me that we return the entity from deleteLibraryCard after we have deleted it. I wonder if it might be better to get it here, then pass it to delete, and then use it, rather than getting it out of the delete method. But if you think this is the most efficient way, we should at least add a comment to deleteLibraryCard in the service class explaining that it returns the entity that has been deleted.